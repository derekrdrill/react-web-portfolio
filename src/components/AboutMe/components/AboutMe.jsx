import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
