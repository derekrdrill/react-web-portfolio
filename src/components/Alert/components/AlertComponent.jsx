import React, { useContext } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = useContext(AlertContext);

  return (
    alert && (
      <StyledAlert severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
