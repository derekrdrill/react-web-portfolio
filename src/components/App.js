import React from 'react';

const AboutMe = React.lazy(() => import('./AboutMe/components/AboutMe'));
const ConnectWithMe = React.lazy(() => import('./ConnectWithMe/components/ConnectWithMe'));
const Homepage = React.lazy(() => import('./Homepage/Homepage'));
const Projects = React.lazy(() => import('./Projects/components/Projects'));

const App = () => (
  <div className='App'>
    <Homepage id='home' />
    <AboutMe id='about-me' />
    <Projects id='projects' />
    <ConnectWithMe id='connect-with-me' />
  </div>
);

export default App;
