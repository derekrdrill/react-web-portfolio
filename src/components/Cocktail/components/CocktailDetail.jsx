import React from 'react';
import styled from 'styled-components';
import { Grid, Typography } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlayCircle } from '@fortawesome/fontawesome-free-solid';

import { CocktailContext } from '../../Cocktail/context/CocktailContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { LoaderSpinner } from '../../LoaderSpinner/LoaderSpinner';

const CocktailDetail = () => {
  const { searchResults, searchResultsLength, loading } = React.useContext(CocktailContext);
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      {loading && <LoaderSpinner open />}
      {!loading && searchResults && searchResults.length > 0 && (
        <CocktailDetailContainer container rowSpacing={2}>
          <Grid item xs={1} md={2} />
          <Grid item xs={10}>
            <Typography variant='h5'>
              {`${searchResultsLength} ${searchResultsLength === 1 ? 'drink' : 'drinks'} listed`}
            </Typography>
          </Grid>
          <Grid item xs={12} />
          <Grid item xs={12} />
          {searchResults.map((searchResult, searchResultKey) => (
            <>
              <CocktailResultsContainer key={searchResultKey} container rowSpacing={4}>
                <Grid item xs={1} md={2} />
                <Grid item xs={11} md={3}>
                  <CocktailImage src={searchResult.strDrinkThumb} />
                </Grid>
                <Grid item xs={1} sm={1} />
                <Grid item xs={10} md={5}>
                  <Typography variant='h3'>{searchResult.strDrink}</Typography>
                  <br />
                  <Grid container spacing={4}>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Ingredients</Typography>
                      {Object.keys(searchResult)
                        .filter(result => result.includes('strIngredient'))
                        .map(
                          (ingredient, ingredientKey) =>
                            searchResult[ingredient] && (
                              <Typography
                                key={`${searchResult.idDrink}${ingredientKey}`}
                                variant='subtitle2'
                              >
                                {`${searchResult[`strMeasure${ingredientKey + 1}`] ?? ''} ${
                                  searchResult[ingredient]
                                }`}
                              </Typography>
                            ),
                        )}
                    </Grid>
                    <Grid item xs={12} sm={6}>
                      <Typography variant='h5'>Glass</Typography>
                      <Typography variant='subtitle2'>{searchResult.strGlass}</Typography>
                    </Grid>
                  </Grid>
                  <br />
                  <Typography variant='h5'>Instructions</Typography>
                  <Typography variant='body2'>{searchResult.strInstructions}</Typography>
                  {searchResult.youtubeData && (
                    <CocktailVideoTutorialLinkContainer item xs={12}>
                      <CocktailVideoTutorialLink
                        href={searchResult.youtubeData.url}
                        target='_blank'
                        rel='noreferrer'
                      >
                        <CocktailVideoTutorial
                          className='youtube-img'
                          src={searchResult.youtubeData.bestThumbnail.url}
                        />
                        <CocktailVideoTutorialPlayButton icon={faPlayCircle} />
                        <CocktailVideoTutorialTitleContainer className='youtube-title'>
                          <CocktailVideoTutorialText variant='body1'>
                            {searchResult.youtubeData.title}
                          </CocktailVideoTutorialText>
                          <CocktailVideoTutorialText variant='subtitle2'>
                            {searchResult.youtubeData.duration}
                          </CocktailVideoTutorialText>
                        </CocktailVideoTutorialTitleContainer>
                      </CocktailVideoTutorialLink>
                    </CocktailVideoTutorialLinkContainer>
                  )}
                </Grid>
                <Grid xs={1} />
                <Grid xs={1} />
                <CocktailResultsDivider
                  item
                  xs={10}
                  darkMode={darkMode}
                  lastItem={searchResultKey + 1 === searchResultsLength}
                />
                <Grid xs={1} />
              </CocktailResultsContainer>
            </>
          ))}
        </CocktailDetailContainer>
      )}
    </>
  );
};

export default CocktailDetail;

export const CocktailDetailContainer = styled(Grid)({
  marginBottom: 150,
  position: 'relative',
  top: 100,
});

export const CocktailImage = styled.img({
  borderRadius: 30,
  height: 275,
  width: 320,
});

export const CocktailResultsContainer = styled(Grid)({
  padding: '40px 0',
});

export const CocktailResultsDivider = styled(Grid)(({ darkMode, lastItem }) => ({
  borderBottom: !lastItem && `solid 1px ${darkMode ? 'gainsboro' : 'grey'}`,
  margin: '0 30px',
  padding: '50px 0',
}));

export const CocktailVideoTutorial = styled.img({
  height: 215,
  width: 315,
});

export const CocktailVideoTutorialLinkContainer = styled(Grid)({
  ':hover': {
    '.youtube-img': {
      opacity: 0.4,
    },
    '.youtube-title': {
      visibility: 'visible',
    },
  },
  height: 215,
  transform: 'translateY(20px)',
  width: 315,
});

export const CocktailVideoTutorialLink = styled.a({
  cursor: 'pointer',
  textDecoration: 'none',
  zIndex: 2,
});

export const CocktailVideoTutorialText = styled(Typography)({
  color: 'white',
  fontWeigt: 'bold',
  zIndex: 0,
});

export const CocktailVideoTutorialTitleContainer = styled.div({
  maxWidth: 300,
  transform: 'translate(10px, -250px)',
  visibility: 'hidden',
});

export const CocktailVideoTutorialPlayButton = styled(FontAwesomeIcon)({
  color: 'red',
  height: 30,
  transform: 'translate(145px, -130px)',
  width: 30,
});
