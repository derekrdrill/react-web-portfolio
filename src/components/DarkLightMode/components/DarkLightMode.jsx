import React, { useContext, useEffect } from 'react';
import styled from 'styled-components';

import { DarkLightModeContext } from '../context/DarkLightModeContext';
import { handleSetDarkMode } from '../context/DarkLightModeActions';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSun, faMoon } from '@fortawesome/fontawesome-free-solid';

export const DarkLightMode = () => {
  const { darkMode, darkLightModeDispatch } = useContext(DarkLightModeContext);

  const callSetHandleDarkMode = () => {
    handleSetDarkMode(darkMode, darkLightModeDispatch);
  };

  useEffect(() => {
    let currentDarkMode = localStorage.getItem('darkMode') === 'true' ? true : false;
    handleSetDarkMode(!currentDarkMode, darkLightModeDispatch);
  }, []);

  return <DarkLightModeIcon darkMode={darkMode} icon={darkMode ? faSun : faMoon} onClick={callSetHandleDarkMode} />;
};

const DarkLightModeIcon = styled(FontAwesomeIcon)(({ darkMode }) => ({
  ':hover': {
    path: {
      fill: darkMode ? '#DDDDDD' : '#759CC9',
    },
  },
  cursor: 'pointer',
  path: {
    fill: darkMode ? '#AAAAAA' : '#759CC9',
  },
}));
