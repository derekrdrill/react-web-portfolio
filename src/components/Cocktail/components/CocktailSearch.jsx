import React from 'react';
import styled from 'styled-components';
import {
  Autocomplete,
  Grid,
  FormControl,
  IconButton,
  InputLabel,
  MenuItem,
  Select,
  TextField,
} from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch } from '@fortawesome/fontawesome-free-solid';

import { AlertContext } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';
import { CocktailContext } from '../../Cocktail/context/CocktailContext';

import { AlertComponent as Alert } from '../../Alert/components/AlertComponent';

import {
  getAllCocktails,
  getAllIngredients,
  getCocktailsByIngredient,
  getSearchOptions,
  handleSearchBarChange,
  setSearchType,
} from '../context/CocktailActions';

const CocktailSearch = () => {
  const { alertDispatch } = React.useContext(AlertContext);

  const {
    cocktailDispatch,
    cocktails,
    ingredients,
    searchResults,
    searchType,
    selectedIngredients,
  } = React.useContext(CocktailContext);

  const { darkMode } = React.useContext(DarkLightModeContext);

  React.useEffect(() => {
    getAllCocktails(cocktailDispatch);
    getAllIngredients(cocktailDispatch);
  }, [cocktailDispatch]);

  return (
    <Grid container>
      <Grid item xs={1} md={2} />
      <CocktailSearchBarGridItem item xs={10} md={8}>
        <CocktailSearchBarContainer>
          <CocktailSearchBar darkMode={darkMode}>
            <InputLabel id='demo-simple-select-label'>Search by...</InputLabel>
            <Select
              value={searchType}
              label='Search by...'
              onChange={e => {
                setSearchType(e, cocktailDispatch, searchType);
              }}
              variant={darkMode ? 'outlined' : 'filled'}
            >
              <MenuItem value='name'>Drink Name</MenuItem>
              <MenuItem value='ingredients'>Ingredients</MenuItem>
            </Select>
          </CocktailSearchBar>
          {searchType === 'name' && (
            <Autocomplete
              freeSolo
              fullWidth
              onChange={(e, searchData) => {
                handleSearchBarChange(cocktailDispatch, searchData, searchType);
              }}
              options={getSearchOptions(searchType, cocktails, ingredients)}
              renderInput={params => (
                <SearchInput
                  {...params}
                  darkMode={darkMode}
                  label='Search by a drink name'
                  variant='filled'
                />
              )}
              value={searchResults ? searchResults[0].strDrink : ''}
            />
          )}
          {searchType == 'ingredients' && (
            <Autocomplete
              freeSolo
              fullWidth
              multiple
              onChange={(e, searchData) => {
                handleSearchBarChange(cocktailDispatch, searchData, searchType);
              }}
              options={getSearchOptions(searchType, cocktails, ingredients)}
              defaultValue={[]}
              renderInput={params => (
                <SearchInput
                  {...params}
                  darkMode={darkMode}
                  label='Select up to 3 ingredients'
                  inputProps={{ ...params.inputProps, maxLength: 3 }}
                  InputProps={{
                    ...params.InputProps,
                    endAdornment: (
                      <SearchInputIconButton
                        onClick={() => {
                          getCocktailsByIngredient(
                            selectedIngredients,
                            cocktailDispatch,
                            alertDispatch,
                          );
                        }}
                      >
                        <FontAwesomeIcon icon={faSearch} />
                      </SearchInputIconButton>
                    ),
                  }}
                  variant='filled'
                />
              )}
              value={selectedIngredients}
            />
          )}
        </CocktailSearchBarContainer>
        <AlertContainer>
          <Alert />
        </AlertContainer>
      </CocktailSearchBarGridItem>
    </Grid>
  );
};

CocktailSearch.propTypes = {};

export default CocktailSearch;

const CocktailSearchBarContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-start',
});

const CocktailSearchBarGridItem = styled(Grid)({
  position: 'relative',
  top: 75,
});

const CocktailSearchBar = styled(FormControl)(({ darkMode }) => ({
  '.MuiInputBase-root': { color: darkMode && 'beige' },
  '.MuiFormLabel-root': {
    color: darkMode && 'beige',
  },
  '.MuiSvgIcon-root': { fill: darkMode && 'beige' },
  backgroundColor: darkMode ? '#414141' : 'white',
  width: 225,
}));

const AlertContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-end',
  marginTop: 10,
});

const SearchInput = styled(TextField)(({ darkMode }) => ({
  '&.MuiFormControl-root': {
    ':hover': {
      backgroundColor: darkMode ? '#607080' : 'lightgrey',
    },
    backgroundColor: darkMode ? '#607080' : 'white',
  },
  '.MuiInputBase-root': {
    borderRadius: 0,
    color: darkMode ? '#75baff' : 'black',
  },
  '.MuiInputLabel-root': {
    color: darkMode ? 'beige' : 'royalblue',
  },
  '.Mui-focused': {
    color: darkMode ? 'beige' : 'black',
    '&.MuiInputBase-root': {
      backgroundColor: darkMode ? '#75baff' : 'white',
    },
  },
  '.MuiChip-label': {
    color: darkMode && 'white',
  },
}));

const SearchInputIconButton = styled(IconButton)({
  transform: 'translateY(-10px)',
});
