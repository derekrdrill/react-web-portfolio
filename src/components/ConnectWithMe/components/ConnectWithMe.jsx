import React, { useContext, useState } from 'react';
import $ from 'jquery';
import styled from 'styled-components';
import { Grid, Typography, Button } from '@mui/material';

import { AlertComponent as Alert } from '../../Alert/components/AlertComponent';
import { DynamicFormInputs } from '../../DynamicFormInputs/DynamicFormInputs';

import { AlertContext } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { handleAlert } from '../../Alert/context/AlertActions';

import { CONNECT_TYPES } from '../constants/CONNECT_TYPES';
import { CONNECT_FORM_INPUTS } from '../constants/CONNECT_FORM_INPUTS';
import bitmojiWaterCooler from '../../../assets/bitmoji_waterCooler.png';

export const ConnectWithMe = ({ id }) => {
  const connectTypes = CONNECT_TYPES;

  const { alertDispatch } = useContext(AlertContext);
  const { darkMode } = useContext(DarkLightModeContext);

  const [firstName, setFirstName] = useState('');
  const [lastName, setLastName] = useState('');
  const [email, setEmail] = useState('');
  const [phone, setPhone] = useState('');
  const [message, setMessage] = useState('');

  const handleMessageSend = async () => {
    if (firstName && lastName && email && message) {
      const response = await fetch(`../send-email/${email}/${firstName}/${lastName}/${message}/${phone}`, {
        method: 'GET',
        headers: { 'Content-Type': 'application/json' },
      }).catch(e => console.warn(e));

      if (response.ok) {
        const messageData = await response.json();
        handleAlert(messageData.message, 'Email sent', 'success', alertDispatch);

        $('#firstName').val('');
        $('#lastName').val('');
        $('#email').val('');
        $('#phone').val('');
        $('#message').val('');
      }
    } else {
      handleAlert(
        'You must enter a first name, last name, email and message',
        'Message cannot send',
        'error',
        alertDispatch,
      );
    }
  };

  const handleInputChange = e => {
    let elementID = e.currentTarget.id;
    let elementValue = e.currentTarget.value;

    if (elementID === 'firstName') {
      setFirstName(elementValue);
    } else if (elementID === 'lastName') {
      setLastName(elementValue);
    } else if (elementID === 'email') {
      setEmail(elementValue);
    } else if (elementID === 'phone') {
      setPhone(elementValue);
    } else if (elementID === 'message') {
      setMessage(elementValue);
    }
  };

  return (
    <ContactPageContainer id={id} darkMode={darkMode} container>
      <TitleContainer item xs={12}>
        <TitleText darkMode={darkMode} variant='h3' component='h2'>
          LET'S CONNECT
        </TitleText>
        <DescriptionText darkMode={darkMode} variant='h6' component='p'>
          Fire up an email, check out my LinkedIn, see this project in Github or send me a message right here!
        </DescriptionText>
      </TitleContainer>
      <ConnectTypeContainer item xs={12} lg={4}>
        <Grid container>
          {connectTypes.map(connect => (
            <Grid key={connect.id} item xs={4} lg={12}>
              <Grid container justifyContent='center'>
                <StyledLink href={connect.href} target={connect.target} darkMode={darkMode}>
                  {connect.icon}
                </StyledLink>
              </Grid>
            </Grid>
          ))}
        </Grid>
      </ConnectTypeContainer>
      <DirectConnectContainer item xs={12} lg={8}>
        <Grid container spacing={2}>
          <Grid item xs={12} lg={5}>
            <DynamicFormInputs
              inputs={CONNECT_FORM_INPUTS[0].inputs}
              onChange={e => {
                handleInputChange(e);
              }}
            />
          </Grid>
          <Grid item xs={12} lg={7}>
            <DynamicFormInputs
              inputs={CONNECT_FORM_INPUTS[1].inputs}
              onChange={e => {
                handleInputChange(e);
              }}
            />
          </Grid>
        </Grid>
        <Grid container justifyContent='space-between'>
          <div>
            <Alert />
          </div>
          <div>
            <Button variant='contained' color='warning' onClick={handleMessageSend}>
              Send
            </Button>
          </div>
        </Grid>
      </DirectConnectContainer>
      <Grid item xs={12}>
        <Grid container justifyContent={{ xs: 'flex-start', lg: 'flex-end' }}>
          <StyledBitmojiImage src={bitmojiWaterCooler} />
        </Grid>
      </Grid>
    </ContactPageContainer>
  );
};

const ContactPageContainer = styled(Grid)(({ darkMode }) => ({
  backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  paddingTop: '0.5%',
  height: '105vh',
}));

const ConnectTypeContainer = styled(Grid)({
  transform: 'translateY(-10px)',
});

const TitleContainer = styled(Grid)({
  paddingRight: '2%',
  display: 'block',
  textAlign: 'right',
});

const TitleText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'beige' : '#759CC9',
  marginBottom: 10,
}));

const DescriptionText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'beige' : '#759CC9',
}));

const StyledLink = styled.a(({ darkMode }) => ({
  svg: {
    width: 110,
    height: 150,
    color: 'beige',
    path: {
      fill: darkMode ? '#36454F' : '#66abc7',
    },
    ':hover': {
      path: {
        fill: darkMode ? '#367993' : '#8fc2d6',
      },
    },
  },
}));

const DirectConnectContainer = styled(Grid)({
  padding: '2% 2% 0 2%',
});

const StyledBitmojiImage = styled.img({
  width: 225,
  height: 240,
  transform: 'translateY(-65px)',
});
