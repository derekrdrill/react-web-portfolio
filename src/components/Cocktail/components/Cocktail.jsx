import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';

import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import CocktailDetail from './CocktailDetail';
import CocktailSearch from './CocktailSearch';

const Cocktail = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <CocktailSearch />
      <CocktailDetail />
    </AlertProvider>
  );
};

Cocktail.propTypes = {};

export default Cocktail;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#1B0008' : '#FBAF93',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
