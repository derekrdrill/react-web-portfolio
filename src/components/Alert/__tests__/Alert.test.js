import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { AlertComponent as Alert, StyledAlert } from '../components/AlertComponent';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('Alert tests', () => {
  let realUseContext;
  let useContextMock;

  beforeEach(() => {
    realUseContext = React.useContext;
    useContextMock = React.useContext = jest.fn();
  });

  afterEach(() => {
    React.useContext = realUseContext;
  });

  it('renders correctly', () => {
    useContextMock.mockReturnValue({ darkMode: true, alert: {}, fadeOut: true });
    renderer.render(<Alert>Title</Alert>);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('renders StyledAlert styled component correctly', () => {
    const alert = shallow(<StyledAlert />);
    const alertFadeOut = shallow(<StyledAlert $fadeOut />);

    expect(alert.props().className).toEqual('sc-bczRLJ eeLUan');
    expect(alertFadeOut.props().className).toEqual('sc-bczRLJ jxwIva');
  });
});
