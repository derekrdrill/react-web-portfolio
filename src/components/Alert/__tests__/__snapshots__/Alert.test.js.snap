// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert tests renders correctly 1`] = `
<Styled(Component)
  $fadeOut={true}
  darkMode={true}
>
  <ForwardRef(AlertTitle) />
</Styled(Component)>
`;
